### PR TITLE
Feature 1.7 convert homo model: server side changes

### PR DESCRIPTION
**Pull request by wfangchi**
Changes:
1. Add converters for HomoLR and HomoNN in federatedml/protobuf/homo_model_convert.
2. Add a "model/homo/convert" rest api in fate_flow that uses the converters.
3. Add unittests for the converters.

fate_client, fate_test and doc changes will be introduced in another change.

### CODE REVIEW

File: python/fate_flow/pipelined_model/publish_model.py [hunk 1]
```diff
@@ -108,3 +111,43 @@ def download_model(request_data):
...
+        if key == 'pipeline':
+            continue
+        for v in value.keys():
+            buffer_obj = model.read_component_model(key, v)
+            module_name = define_index.get("component_define", {}).get(key, {}).get('module_name')
+            converted_framework, converted_model = model_convert(model_contents=buffer_obj,
+                                                                 module_name=module_name,
+                                                                 framework_name=framework_name)
+            if converted_model:
+                converted_model_dir = os.path.join(model.variables_data_path, key, "converted_model")
```
Comment (jarviszeng-zjc) [inline python/fate_flow/pipelined_model/publish_model.py:137]: The path  missing model_alias

File: python/fate_flow/pipelined_model/publish_model.py [hunk 2]
```diff
@@ -108,3 +111,43 @@ def download_model(request_data):
...
+                os.makedirs(converted_model_dir, exist_ok=True)
+
+                saved_path = save_converted_model(converted_model,
+                                                  converted_framework,
+                                                  converted_model_dir)
+                detail.append({
+                    "component_name": key,
+                    "converted_model_path": saved_path
+                })
```
Comment (jarviszeng-zjc) [inline python/fate_flow/pipelined_model/publish_model.py:146]: model alias

File: python/fate_flow/apps/model_app.py [hunk 3]
```diff
@@ -726,6 +726,20 @@ def get_predict_conf():
     return error_response(210, "No model found, please check if arguments are specified correctly.")
 
 
+@manager.route('/homo/convert', methods=['POST'])
+def homo_convert():
+    request_config = request.json or request.form.to_dict()
+    required_arguments = ["model_id", "model_version", "role", "party_id"]
+    check_config(request_config, required_arguments=required_arguments)
+    request_config["model_id"] = gen_party_model_id(model_id=request_config["model_id"],
```
[the remark this answers is not in the document]
Comment (wfangchi) [reply python/fate_flow/apps/model_app.py]: Okay. Updated.

File: python/fate_flow/pipelined_model/publish_model.py (hunk 1 above)
Comment (wfangchi) [reply python/fate_flow/pipelined_model/publish_model.py:139]: I was thinking "model_alias" is "converted_model" - at the same level as "model". So the folder layout looks like

```
variables/data/homo_lr_0/
├── converted_model
│   └── sklearn.joblib
└── model
    ├── HomoLogisticRegressionMeta
    └── HomoLogisticRegressionParam
```

Could you help see if that is ok?

Comment (jarviszeng-zjc) [reply python/fate_flow/pipelined_model/publish_model.py:139]: variables
├── data
│   └── dataio_0
│       └── dataio
│           ├── DataIOMeta
│           └── DataIOParam

and refer https://github.com/FederatedAI/FATE/blob/6c5263d8d1ed0423884fda10dd30304cd6d25f8d/python/fate_flow/pipelined_model/pipelined_model.py#L61

Comment (wfangchi) [reply python/fate_flow/pipelined_model/publish_model.py]: Got it. I moved the folder under the model_alias folder now.

File: python/fate_flow/pipelined_model/publish_model.py (hunk 2 above)
Comment (wfangchi) [reply python/fate_flow/pipelined_model/publish_model.py:149]: Added the model_alias field.

**Review by jarviszeng-zjc — APPROVED** [review]

**Review by weiwee — APPROVED** [review]

File: python/federatedml/protobuf/homo_model_convert/pytorch/nn.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    def convert(self, model_dict):
+        param_obj = model_dict["HomoNNModelParam"]
+        meta_obj = model_dict["HomoNNModelMeta"]
+        if meta_obj.params.config_type != "pytorch":
+            raise ValueError("Invalid config type: {}".format(meta_obj.config_type))
+
+        with io.BytesIO(param_obj.saved_model_bytes) as model_bytes:
+            if hasattr(param_obj, "api_version") and param_obj.api_version > 0:
+                from ....nn.homo_nn._torch import FedLightModule
```
Comment (mgqa34) [inline python/federatedml/protobuf/homo_model_convert/pytorch/nn.py:37]: use federatedml.nn.homo_nn._torch maybe betther

Comment (wfangchi) [reply python/federatedml/protobuf/homo_model_convert/pytorch/nn.py]: Good suggestion. Updated.

**Review by mgqa34 — APPROVED** [review]

**Pull request merged**